### PR TITLE
[docs] Editing pass on clusters docs, removing legacy material and fixing style issues

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

- Fix remaining placeholder items and missing transition material from cluster docs
- Remove legacy materials from clusters / ML libraries
- Fix link refs and styling
- General editing passes and minor reorganization

### CODE REVIEW

File: doc/source/_toc.yml [hunk 1]
```diff
@@ -296,25 +292,13 @@ parts:
...
-        - file: cluster-deprecated/usage-stats
-        - file: cluster-deprecated/user-guide
-        - file: cluster-deprecated/cluster_under_construction.md
-        - file: cluster-deprecated/deploy
-        - file: cluster-deprecated/api
 
   - caption: References
     chapters:
       - file: ray-references/api
+      - file: usage-stats
```
Comment (cadedaniel) [inline doc/source/_toc.yml:301]: Warning: [hardcoded references](https://github.com/ray-project/ray/blob/master/python/ray/_private/usage/usage_constants.py#L21) to `https://docs.ray.io/en/master/cluster/usage-stats.html` exist.

File: doc/source/cluster/getting-started.rst [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
-
+.. _cluster-index:
 
 Ray Clusters Overview
 =====================
 
-What is a Ray cluster?
-----------------------
+Ray enables seamless scaling of workloads from a laptop to a large cluster. While Ray
+works out of the box on single machines with just a call to `ray.init`, to run Ray
```
Comment (stephanie-wang) [inline doc/source/cluster/getting-started.rst:7]:
```suggestion
works out of the box on single machines with just a call to ``ray.init``, to run Ray
```

File: doc/source/cluster/getting-started.rst [hunk 3]
```diff
@@ -1,36 +1,29 @@
...
 
-* The :ref:`Ray cluster launcher on AWS and GCP<ref-cluster-quick-start-vms-under-construction>`. Community-supported Azure and Aliyun integrations also exist.
-* :ref:`KubeRay, the official way to run Ray on Kubernetes<kuberay-index>`.
+* On :ref:`AWS and GCP <cloud-vm-index>`. Community-supported Azure and Aliyun integrations also exist.
+* On :ref:`Kubernetes <kuberay-index>`, via the officially supported KubeRay project.
 
-Advanced users may want to :ref:`deploy Ray clusters on-premise <on-prem>`
-or onto infrastructure platforms not listed here by :ref:`providing a custom node provider <ref-cluster-setup-under-construction>`.
+Advanced users may want to :ref:`deploy Ray manually <on-prem>`
+or onto platforms not listed here by :ref:`implementing a custom node provider <ref-cluster-setup>`.
```
Comment (stephanie-wang) [inline doc/source/cluster/getting-started.rst:23]:
```suggestion
or onto :ref:`platforms not listed here <ref-cluster-setup>`.
```
(at this point we don't know what a node provider is)

File: doc/source/cluster/getting-started.rst [hunk 4]
```diff
@@ -42,34 +35,35 @@ Where to go from here?
...
+    Take a sample application designed to run on a laptop and scale it up in the
+    cloud. Access to an AWS or GCP account is required.
 
     +++
-    .. link-button:: kuberay-quickstart
+    .. link-button:: vm-cluster-quick-start
         :type: ref
-        :text: Getting Started with Ray on Kubernetes
+        :text: Get Started with Ray on VMs
         :classes: btn-outline-info btn-block
```
Comment (stephanie-wang) [inline doc/source/cluster/getting-started.rst:68]: Forgot to do this earlier: add a card like "I want to run my application on an existing Ray cluster" and link to jobs?

File: doc/source/cluster/key-concepts.rst [hunk 5]
```diff
@@ -1,77 +1,66 @@
...
 
-To learn more about the autoscaler and how to configure it, refer to the following user guides:
+The *Ray autoscaler* is a process that runs on the :ref:`head node <cluster-head-node>` (or as a sidecar container in the head pod if :ref:`using Kubernetes <kuberay-index>`).
+When the resource demands of the Ray workload exceed the
+current capacity of the cluster, the autoscaler will try to increase the number of worker nodes. When worker nodes
+sit idle, the autoscaler will remove worker nodes from the cluster.
 
-* :ref:`Configuring Autoscaling on VMs<deployment-guide-autoscaler-under-construction>`.
-* :ref:`Autoscaling on Kubernetes<kuberay-autoscaler-discussion>`.
+It is important to understand that the autoscaler only reacts to task and actor resource requests, and not application metrics.
```
Comment (stephanie-wang) [inline doc/source/cluster/key-concepts.rst:60]:
```suggestion
It is important to understand that the autoscaler only reacts to task and actor resource requests, and not application metrics or physical resource utilization.
```

File: doc/source/cluster/key-concepts.rst [hunk 6]
```diff
@@ -81,4 +70,9 @@ Ray Jobs enable users to submit locally developed-and-tested applications to a
 remote Ray cluster. Ray Job Submission simplifies the experience of packaging,
 deploying, and managing a Ray application.
 
-To learn how to run workloads on a Ray Cluster, refer to the :ref:`Ray Jobs guide<ray-jobs-under-construction>`.
+For interactive development, the following additional methods are available:
```
Comment (stephanie-wang) [inline doc/source/cluster/key-concepts.rst:73]: Nice!

File: doc/source/cluster/key-concepts.rst [hunk 7]
```diff
@@ -81,4 +70,9 @@ Ray Jobs enable users to submit locally developed-and-tested applications to a
...
+
+* Directly running a script or notebook on any head or worker node.
+* Using the Ray Client to connect remotely to the cluster.
+
+To learn how to run workloads on a Ray Cluster, refer to the :ref:`Ray Jobs guide <jobs-overview>`.
```
Comment (stephanie-wang) [inline doc/source/cluster/key-concepts.rst:78]:
```suggestion
To learn how to run workloads on a Ray cluster, refer to the :ref:`Ray Jobs guide <jobs-overview>`.
```

File: doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md [hunk 8]
```diff
@@ -1,36 +1,43 @@
...
-workload will require, autoscaling can adjust your Ray cluster to the right size.
+Note that the autoscaler only considers logical resource requests for scaling (i.e., those specified in ``@ray.remote`` and displayed in `ray status`), not physical machine utilization. If a user tries to launch an actor, task, or placement group but there are insufficient resources, the request will be queued. The autoscaler adds nodes to satisfy resource demands in this queue.
+The autoscaler also removes nodes after they become idle for some time.
+A node is considered idle if it has no active tasks, actors, or objects.
 
-**Save on costs.** Idle compute is automatically scaled down, potentially leading to cost savings,
-especially when you are using expensive resources like GPUs.
+```{admonition} When to use Autoscaling?
+Autoscaling can reduce workload costs, but adds node launch overheads and can be tricky to configure.
+We recommend starting with non-autoscaling clusters if you're new to Ray.
```
Comment (stephanie-wang) [inline doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md:15]: Unfortunately I think all of our quickstart examples and the cluster launcher's defaults do use autoscaling... maybe add a TODO to switch out the first examples with fixed-size configs.

File: doc/source/cluster/running-applications/job-submission/quickstart.rst [hunk 9]
```diff
@@ -1,17 +1,11 @@
...
-.. _jobs-quickstart-under-construction:
+.. _jobs-quickstart:
 
 =================================
 Quickstart Using the Ray Jobs CLI
 =================================
 
 In this guide, we will walk through the Ray Jobs CLIs available for submitting and interacting with a Ray Job.
 
-.. note::
```
Comment (stephanie-wang) [inline doc/source/cluster/running-applications/job-submission/quickstart.rst:12]: Why the removal? Jobs is still in beta.

File: doc/source/cluster/vms/index.md [hunk 10]
```diff
@@ -1,4 +1,71 @@
...
+(cloud-vm-index)=
+
+## Overview
+
+In this section we cover how launch Ray clusters on Cloud VMs. Ray ships with built-in support
+for launching AWS and GCP clusters, and also has community integrations for Azure and Aliyun.
+Each Ray cluster consists of a head node pod and a collection of worker node pods. Optional
+autoscaling support allows the Ray cluster to be sized according to the
+requirements of your Ray workload, adding and removing worker nodes as needed. Ray supports
+clusters composed of multiple heterogenous compute nodes (including GPU nodes).
```
Comment (stephanie-wang) [inline doc/source/cluster/vms/index.md:11]:
```suggestion
In this section we cover how to launch Ray clusters on Cloud VMs. Ray ships with built-in support
for launching AWS and GCP clusters, and also has community-maintained integrations for Azure and Aliyun.
Each Ray cluster consists of a head node and a collection of worker nodes. Optional
[autoscaling](vms-autoscaling) support allows the Ray cluster to be sized according to the
requirements of your Ray workload, adding and removing worker nodes as needed. Ray supports
clusters composed of multiple heterogenous compute nodes (including GPU nodes).
```

File: doc/source/cluster/vms/index.md [hunk 11]
```diff
@@ -1,4 +1,71 @@
...
+
+Concretely, you will learn how to:
+
+- Set up and configure Ray in public clouds
+- Deploy applications and monitor your cluster
+
+## Learn More
+
+The Ray docs present all the information you need to start running Ray workloads on Kubernetes.
```
Comment (stephanie-wang) [inline doc/source/cluster/vms/index.md:20]:
```suggestion
The Ray docs present all the information you need to start running Ray workloads on VMs.
```

File: doc/source/cluster/vms/user-guides/configuring-autoscaling.rst [hunk 12]
```diff
@@ -1,51 +1,56 @@
...
+`idle_timeout_minutes[default_value=5, min_value=0]`: The number of minutes that need to pass before an idle worker node is removed by the
+autoscaler. The smaller the value, the more aggressive downscaling will be. Worker nodes are considered idle when they hold no active tasks, actors, or referenced objects (either in-memory or spilled to disk). This parameter does not affect the head node.
 
-Autoscaler SDK
-==============
+Programmatic Scaling
+--------------------
 
-For more information on programmatic access to the autoscaler, see :ref:`Autoscaler SDK<ref-autoscaler-sdk-under-construction>`.
+For more information on programmatic access to the autoscaler, see :ref:`Programmatic Cluster Scaling Guide <ref-autoscaler-sdk>`.
```
Comment (stephanie-wang) [inline doc/source/cluster/vms/user-guides/configuring-autoscaling.rst:56]:
```suggestion
For more information on programmatic access to the autoscaler, see the :ref:`Programmatic Cluster Scaling Guide <ref-autoscaler-sdk>`.
```

File: doc/source/ray-air/key-concepts.rst [hunk 13]
```diff
@@ -16,14 +16,11 @@ Datasets
 
 
 Preprocessors
```
Comment (stephanie-wang) [inline doc/source/ray-air/key-concepts.rst:18]: Were these changes meant to be included?

**Review by stephanie-wang — APPROVED** [review]
Mainly grammar fixes.

It does look like some changes got included from a different PR, FYI.

File: doc/source/_toc.yml (hunk 1 above)
Comment (ericl) [reply doc/source/_toc.yml:301]: Thanks for the reminder; moved it back.

File: doc/source/cluster/running-applications/job-submission/quickstart.rst (hunk 9 above)
Comment (ericl) [reply doc/source/cluster/running-applications/job-submission/quickstart.rst:12]: Reverted this. Though, it is very odd we're recommending a beta.

File: doc/source/ray-air/key-concepts.rst (hunk 13 above)
Comment (ericl) [reply doc/source/ray-air/key-concepts.rst:18]: Yeah, it's a bit of a drive by to fix some minor issues.

**Review by simon-mo — APPROVED** [review]
stamp Serve ref changes

File: doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md [hunk 14]
```diff
@@ -88,7 +96,73 @@ $ kubectl logs raycluster-autoscaler-head-xxxxx -c autoscaler | tail -n 20
...
+    - `Conservative`: Upscaling is rate-limited; the number of pending worker pods is at most the number
+      of worker pods connected to the Ray cluster.
+    - `Default`: Upscaling is not rate-limited.
+    - `Aggressive`: An alias for Default; upscaling is not rate-limited.
+
+**`idleTimeoutSeconds`** (default 60s): This is the number of seconds to wait before scaling down an idle worker pod. Worker nodes are considered idle when they hold no active tasks, actors, or referenced objects (either in-memory or spilled to disk).
+
+### Configuring the autoscaler sidecar container
+
+The `autoscalerOptions` also provides some options for configuring the autoscaler itself. Usually, it should not be necessary to modify these options.
```
Comment (DmitriGekhtman) [inline doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md:133]:
```suggestion
The `autoscalerOptions` field also provides options for configuring the autoscaler container. Usually, it is not necessary to specify these options.
```

File: doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md [hunk 15]
```diff
@@ -88,7 +96,73 @@ $ kubectl logs raycluster-autoscaler-head-xxxxx -c autoscaler | tail -n 20
...
+
+**`resources`**: The `resources` subfield of `autoscalerOptions` sets optional resource overrides
+for the autoscaler sidecar container. These overrides
+should be specified in the standard [container resource
+spec format](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#resources).
+The default values are as indicated below:
```
Comment (DmitriGekhtman) [inline doc/source/cluster/kubernetes/user-guides/configuring-autoscaling.md:139]: picking at my own wording
```suggestion
The default values are indicated below:
```

**Pull request merged**